app/controllers/profiles_controller.rb

```diff
@@ -1,16 +1,19 @@
 class ProfilesController < ApplicationController
-  before_action :find_profile, except: %i[show]
+  before_action :find_profile, only: %i[show edit update]
 
-  def show
-    @profile = Profile.includes(:user, :picture_blob).friendly.find(params[:id])
-  end
+  def show; end
 
   def edit; end
 
   def update
-    if @profile.update(profile_params)
+    update_records = ActiveRecord::Base.transaction do
+      @profile.update!(profile_params)
+      Resumes::Update.new(user: current_user, params: resume_params).call!
+    end
+    if update_records
       redirect_to @profile, success: 'Profile successfully updated!'
     else
+      # using this now misses validations and errors on the resume part of the form.
       flash.now[:form_errors] = @profile.errors.full_messages
       render :edit, status: :unprocessable_entity
     end
```

Comment:
I feel like it should always just redirect and the `else` can just happen if there is an error (i.e: rescue).

Reply:
Can you clarify this, please?

Reply:
this service isn't going to be returning anything. Wouldn't it just be `nil` or raising some type of error which you'd need to capture in the controller?

Reply:
got it. yeah, fair enough.